src/lib.rs

```diff
@@ -1028,16 +1030,18 @@ impl<'a, K: 'a + Eq + Hash, V: 'a, S: 'a + BuildHasher + Clone> Map<'a, K, V, S>
         K: Borrow<Q>,
         Q: Hash + Eq + ?Sized,
     {
-        let hash = self.hash_usize(&key);
+        let hash = self.hash_u64(&key);
 
-        let idx = self.determine_shard(hash);
+        let idx = self.determine_shard(hash as usize);
 
-        let shard = unsafe { self._yield_write_shard(idx) };
+        let mut shard = unsafe { self._yield_write_shard(idx) };
 
-        if let Some((kptr, vptr)) = shard.get_key_value(key) {
+        if let RawEntryMut::Occupied(mut entry) =
+            shard.raw_entry_mut().from_key_hashed_nocheck(hash, key)
+        {
             unsafe {
-                let kptr: *const K = kptr;
-                let vptr: *mut V = vptr.as_ptr();
+                let kptr: *const K = entry.key();
+                let vptr: *mut V = entry.get_mut().as_ptr();
```

Comment:
I believe this violates stacked borrows, the `entry.get_mut()` invalidates the `*const` to the key. Inverting the order of the operations should be enough though.